server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/TransportGetAliasesAction.java

```diff
@@ -147,20 +146,17 @@ static Map<String, List<DataStreamAlias>> postProcess(
         ClusterState state
     ) {
         Map<String, List<DataStreamAlias>> result = new HashMap<>();
-        boolean noAliasesSpecified = request.getOriginalAliases() == null || request.getOriginalAliases().length == 0;
         List<String> requestedDataStreams = resolver.dataStreamNames(state, request.indicesOptions(), request.indices());
+
+        Map<String, List<DataStreamAlias>> dsAliases = state.metadata()
+            .findDataStreamAliases(request.aliases(), requestedDataStreams.toArray(new String[0]));
+
         for (String requestedDataStream : requestedDataStreams) {
-            List<DataStreamAlias> aliases = state.metadata()
-                .dataStreamAliases()
-                .values()
-                .stream()
-                .filter(alias -> alias.getDataStreams().contains(requestedDataStream))
-                .filter(alias -> noAliasesSpecified || Regex.simpleMatch(request.aliases(), alias.getName()))
-                .toList();
-            if (aliases.isEmpty() == false) {
-                result.put(requestedDataStream, aliases);
+            if (dsAliases.containsKey(requestedDataStream)) {
```

Comment:
I think that this check is redundant now, right? In Metadata, the setter only puts things in the map that's returned if they are the things you passed in as the second argument to `findDataStreamAliases`.

Reply:
That is, can you just return the result of `findDataStreamAliases` from this method?